File: app/src/main/java/com/josegonzalez/musicsearchsample/view/itemList/ItemListView.java
package com.josegonzalez.musicsearchsample.view.itemList;

import com.josegonzalez.musicsearchsample.model.TrackListDto;

public interface ItemListView {

    void addTrackList(TrackListDto listDto);

    void onError();

    void onLoading();

    void showDetails(String artistName, String trackName, String imageUrl);

}

File: app/src/main/java/com/josegonzalez/musicsearchsample/features/list/ItemListView.kt
package com.josegonzalez.musicsearchsample.features.list

import com.josegonzalez.musicsearchsample.model.TrackListDto

interface ItemListView {

    fun addTrackList(listDto: TrackListDto?)
    fun onError()
    fun onLoading()
    fun showDetails(artistName: String?, trackName: String?, imageUrl: String?)

}